honeycombextension/extension.go

```diff
@@ -181,23 +204,16 @@ func (h *honeycombExtension) reportUsage() {
 				continue
 			}
 
-			sendingChan, err := h.telemetryHandler.SendMessage(reportUsageMessageType, data)
-			switch {
-			case err == nil:
-				// Count successful report sent
-				h.telemetryBuilder.HoneycombExtensionUsageReportSuccess.Add(context.Background(), 1)
-			case errors.Is(err, types.ErrCustomMessagePending):
-				h.telemetryBuilder.HoneycombExtensionUsageReportPending.Add(context.Background(), 1)
-				<-sendingChan
-				continue
-			default:
-				h.telemetryBuilder.HoneycombExtensionUsageReportFailure.Add(context.Background(), 1)
-				h.set.Logger.Error("failed to send message", zap.Error(err))
+			shouldRetry := h.sendUsageReport(data)
+			// If the message was pending, wait for it to complete and retry once
+			if shouldRetry {
+				h.set.Logger.Debug("Pending message completed, retrying once")
+
+				h.sendUsageReport(data)
```

Comment:
Lets capture some telemetry if we get another pending message. Probably another `Error` log and increment `HoneycombExtensionUsageReportFailure`.

Reply:
I added a `Error` log for retry failure case. The `HoneycombExtensionUsageReportFailure` is already incremented inside the `sendUsageReport`